lib/time/time.go

```diff
@@ -71,8 +71,25 @@ var Module = &starlarkstruct.Module{
 // NowFunc is a function that generates the current time. Intentionally exported
 // so that it can be overridden, for example by applications that require their
 // Starlark scripts to be fully deterministic.
+//
+// Deprecated: use SetNow() to set per-thread clock.
 var NowFunc = time.Now
 
+const contextKey = "time.Now"
+
+// SetNow associates with the specified Starlark thread a function that generates the current time.
+// Can be used for example by applications that require their
+// Starlark scripts to be fully deterministic.
```

Comment:
Let's phrase this as:

// SetNow sets the thread's optional clock function.
// If non-nil, it will be used in preference to NowFunc when the
// thread requests the current time by executing a call to time.now.